package.json

```diff
@@ -148,10 +148,12 @@
     "koa-socket-2": "^1.0.17",
     "lru-cache": "^4.1.1",
     "prop-types": "^15.5.10",
+    "regenerator-runtime": "^0.13.3",
```

Comment:
Why is `regenerator-runtime` being added?

Reply:
The build process was erring out without. Not sure what exactly requires it.

Reply:
Hi @ksjogo.  Can you double-check this again?

What's the error message that you get when regenerator-runtime is not added?

Reply:
Without that line I get these errors in a gatsby project depending on boardgame.io
```
ERROR #98123  WEBPACK

Generating JavaScript bundles failed

Can't resolve 'regenerator-runtime/runtime' in '/Users/johannes/dev/ksjogo/packages/boardgame.io/dist'

File: ../../packages/boardgame.io/dist/boardgameio.es.js


 ERROR #98123  WEBPACK

Generating JavaScript bundles failed

Can't resolve 'regenerator-runtime/runtime' in '/Users/johannes/dev/ksjogo/packages/boardgame.io/dist/esm'

File: ../../packages/boardgame.io/dist/esm/ai-601d4764.js


 ERROR #98123  WEBPACK

Generating JavaScript bundles failed

Can't resolve 'regenerator-runtime/runtime' in '/Users/johannes/dev/ksjogo/packages/boardgame.io/dist/esm'

File: ../../packages/boardgame.io/dist/esm/Debug-f72fc892.js

not finished run queries - 13.777s
```

Reply:
Is it possible that this is due to the way Gatsby's webpack config is set up?

The reason I'm picky about this is:

1. We don't transform async functions to generators: [source](https://github.com/nicolodavis/boardgame.io/blob/master/babel.config.js#L7)

2. A simple grep in `dist/` for `regenerator` returns no results.

So, my understanding is that boardgame.io has no need for `regenerator-runtime` (it just leaves async functions untouched).  What I suspect is happening is that the Babel config in your client project is transforming async functions and that's why you see the error (which can probably be fixed easy by adding `regenerator-runtime` as a dep to your Gatsby project.

Can you try this?

Reply:
For what it’s worth, I’ve just run into the same error when trying to create a plain JS version of the TicTacToe multiplayer example: https://codesandbox.io/s/bgio-plain-js-multiplayer-server-742oh

Looks OK on CodeSandbox (until you click and it fails to connect to the server), but when downloaded and served using Parcel I see `ReferenceError: regeneratorRuntime is not defined` in the browser console.

Reply:
This does appear to be Babel-related. There’s some more info here: https://github.com/babel/babel/issues/9849 but those suggestions don’t seem to help and replicating the `"exclude": ["transform-regenerator", "transform-async-to-generator"]` config boardgame.io uses, also didn’t help.

The only thing that does work is setting `browserslist` to [`> 0.4% and not ie 11`](https://browserl.ist/?q=%3E+0.4%25+and+not+ie+11) (excluding IE11 & other browsers that don’t support async).

Reply:
Might it be the case that your dist has rolled regenerator runtime into your dist bundle and mine hasn't?
Will have another look.

Reply:
@ksjogo No, this is a problem with conflicting build targets. boardgame.io builds without transforming async functions so regenerator runtime isn't needed. However if you try to compile boardgame.io to a different set of browser targets, Babel will transform async functions to support older browsers (IE11 and a few others). Not sure what the solution here would be other than indicating no support for those older browsers.